src/derivative_utils.jl

```diff
@@ -0,0 +1,84 @@
+function calc_tderivative!(integrator, cache, repeat_step)
+  @inbounds begin
+    @unpack t,dt,uprev,u,f,p = integrator
+    @unpack du2,fsalfirst,dT,tf,linsolve_tmp = cache
+    d1 = cache.tab.d1
+    dtd1 = dt*d1
+
+    # Time derivative
+    if !repeat_step # skip calculation if step is repeated
+      if has_tgrad(f)
+        f(Val{:tgrad}, dT, uprev, p, t)
+      else
+        tf.uprev = uprev
+        derivative!(dT, tf, t, du2, integrator)
+      end
+    end
+
+    f( fsalfirst,  uprev, p, t)
+
+    @. linsolve_tmp = fsalfirst + dtd1*dT
+
+  end
+end
+
+function calc_jacobian!(integrator, cache, repeat_step)
+  @inbounds begin
+    is_2332 = typeof(cache) <: Rosenbrock23Cache || typeof(cache) <: Rosenbrock32Cache
+    @unpack t,dt,uprev,u,f,p = integrator
+    @unpack du1,du2,fsalfirst,dT,J,W,uf,tf,linsolve_tmp,linsolve_tmp_vec,jac_config,vectmp = cache
+    d1 = cache.tab.d1
```

Comment:
Should refactor the SDIRK methods at the same time. I don't think this trick of relying on the same names here will actually carry over well, and we'll want this to be for all stiff solvers as much as possible. I think there may be some things that need to get passed around.